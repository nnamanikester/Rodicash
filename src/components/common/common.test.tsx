import * as React from 'react';
import {render} from '@testing-library/react-native';
import * as UI from '.';

describe('Common Unit Components', () => {
  it('Should export "Block" component', () => {
    const Block = render(<UI.Block />);
    expect(Block).toMatchSnapshot();
  });

  it('Should export "Button" component', () => {
    const Button = render(<UI.Button />);
    expect(Button).toMatchSnapshot();
  });

  it('Should export "Clickable" component', () => {
    const Clickable = render(<UI.Clickable />);
    expect(Clickable).toMatchSnapshot();
  });

  it('Should export "Column" component', () => {
    const Column = render(<UI.Column />);
    expect(Column).toMatchSnapshot();
  });

  it('Should export "Icon" component', () => {
    const Icon = render(<UI.Icon name="md-menu" />);
    expect(Icon).toMatchSnapshot();
  });

  it('Should export "Layout" component', () => {
    const Layout = render(<UI.Layout />);
    expect(Layout).toMatchSnapshot();
  });

  it('Should export "Link" component', () => {
    const Link = render(<UI.Link />);
    expect(Link).toMatchSnapshot();
  });

  it('Should export "Loading" component', () => {
    const Loading = render(<UI.Loading show={false} />);
    expect(Loading).toMatchSnapshot();
  });

  it('Should export "Row" component', () => {
    const Row = render(<UI.Row />);
    expect(Row).toMatchSnapshot();
  });

  it('Should export "PinInput" component', () => {
    const PinInput = render(<UI.PinInput />);
    expect(PinInput).toMatchSnapshot();
  });

  it('Should export "Spacer" component', () => {
    const Spacer = render(<UI.Spacer />);
    expect(Spacer).toMatchSnapshot();
  });

  it('Should export "Text" component', () => {
    const Text = render(<UI.Text />);
    expect(Text).toMatchSnapshot();
  });

  it('Should export "TextInput" component', () => {
    const TextInput = render(<UI.TextInput />);
    expect(TextInput).toMatchSnapshot();
  });
});
