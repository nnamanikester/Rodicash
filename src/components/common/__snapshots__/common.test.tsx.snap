// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Unit Components Should export "Block" component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 0,
        "height": "auto",
        "position": "relative",
        "width": "100%",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`Common Unit Components Should export "Button" component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294967295,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 66.5,
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Common Unit Components Should export "Clickable" component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;

exports[`Common Unit Components Should export "Column" component 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
/>
`;

exports[`Common Unit Components Should export "Icon" component 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#8AA3B9",
        "fontSize": 28,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ÔèΩ
</Text>
`;

exports[`Common Unit Components Should export "Layout" component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={true}
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "#3AC295",
          ]
        }
        onRefresh={[Function]}
        progressBackgroundColor="#fff"
        refreshing={false}
      />
    }
    scrollEventThrottle={400}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "paddingHorizontal": 37.5,
        },
        Object {},
      ]
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;

exports[`Common Unit Components Should export "Link" component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "auto",
    }
  }
>
  <Text
    bold={true}
    color="#D0981B"
    style={
      Array [
        Object {
          "color": "#0E253A",
          "fontFamily": "Gordita-Regular",
          "fontSize": 22.5,
          "lineHeight": 32,
        },
        Object {
          "color": "#D0981B",
          "fontFamily": "Gordita-Bold",
          "fontSize": 20,
          "lineHeight": 32,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Common Unit Components Should export "Loading" component 1`] = `null`;

exports[`Common Unit Components Should export "PinInput" component 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  nativeID="animatedComponent"
  style={
    Object {
      "alignItems": "stretch",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "position": "relative",
      "width": 675,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "100%",
        "margin": 0,
        "position": "absolute",
      }
    }
  >
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#D4E1EC",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 93.5,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "width": 75,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0E253A",
              "fontSize": 40,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <TextInput
    autoFocus={false}
    caretHidden={true}
    disableFullscreenUI={true}
    editable={true}
    keyboardType="number-pad"
    maxLength={6}
    numberOfLines={1}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    selection={
      Object {
        "end": 0,
        "start": 0,
      }
    }
    spellCheck={false}
    style={
      Object {
        "flex": 1,
        "opacity": 0,
        "textAlign": "center",
      }
    }
    value=""
  />
</View>
`;

exports[`Common Unit Components Should export "Row" component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "width": "100%",
    }
  }
/>
`;

exports[`Common Unit Components Should export "Spacer" component 1`] = `
<View
  style={
    Object {
      "margin": 8,
    }
  }
/>
`;

exports[`Common Unit Components Should export "Text" component 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0E253A",
        "fontFamily": "Gordita-Regular",
        "fontSize": 22.5,
        "lineHeight": 32,
      },
      Object {
        "color": "#0E253A",
      },
      undefined,
    ]
  }
/>
`;

exports[`Common Unit Components Should export "TextInput" component 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8AA3B9"
      style={
        Object {
          "backgroundColor": "#F4F6FA",
          "borderColor": "#3AC295",
          "borderRadius": 10,
          "borderWidth": 0,
          "color": "#0E253A",
          "flex": 1,
          "fontFamily": "Gordita-Regular",
          "fontSize": 24,
          "height": 80,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      textBreakStrategy="highQuality"
    />
  </View>
</View>
`;
